### PR TITLE
common/clock: right restart time.Timer

### PR DESCRIPTION
**Pull request by 0xcb9ff9**
### Description

https://github.com/golang/go/issues/14038

https://forum.golangbridge.org/t/the-right-way-to-restart-time-timer/17472

create clock package, right restart `time.Timer`

if `rescheduleFetch` reset time too small, block `fetchTimer` next trigger

https://github.com/dogechain-lab/dbsc/blob/96778f1fc3647e466ec966c16935400bf1fa408c/eth/fetcher/block_fetcher.go#L427-L429

https://github.com/dogechain-lab/dbsc/blob/96778f1fc3647e466ec966c16935400bf1fa408c/eth/fetcher/block_fetcher.go#L530-L531

### CODE REVIEW

File: common/clock/clock.go
Comment (DarianShawn) [inline common/clock/clock.go:1]: Any tests on self-define timer?

**Review by DarianShawn — COMMENTED** [review]
The PR is good to go if we provide more tests on the new interface.

File: common/clock/clock.go
Comment (0xcb9ff9) [reply common/clock/clock.go:1]: done [05c7420](https://github.com/dogechain-lab/dbsc/pull/29/commits/05c74202e561e78ccece5a84d6da1e7102a93b81)

**Review by DarianShawn — APPROVED** [review]
👍🏻
`nocopy` is very important statement of `Clock` type.

No more questions now.

**Pull request merged**